src/lib/mdns/platform/Mdns.h

```diff
@@ -129,7 +129,18 @@ CHIP_ERROR ChipMdnsInit(MdnsAsyncReturnCallback initCallback, MdnsAsyncReturnCal
 CHIP_ERROR ChipMdnsShutdown();
 
 /**
- * This function publishes an service via mDNS.
+ * Removes or marks all services being advertised for removal.
+ *
+ * Depending on the implementation, the function may either stop advertising existing services
+ * immediately, or mark them for removal upon the subsequent `ChipMdnsCompleteServiceUpdate` call.
+ *
+ * @retval CHIP_NO_ERROR  The removal succeeds.
+ * @retval Error code     The removal fails.
+ */
+CHIP_ERROR ChipMdnsRemoveServices();
```

Comment:
These names are all "mDNS" cast.

All of this would be easier with:

- namespace `chip::dnssd` rather than `chip::Mdns`
- Replace `kMdns*` with `kDnsSd*`
- Replace prefix of `ChipMdns` with nothing

This could be done in a follow-up, but this odd smurf naming when already in a namespace has the following bad side-effects:

- Adds `chip` to the name of functions (when Matter is the name, which could be renamed by namespace rename of `chip::` to `matter::` in the future
- Ties to "mDNS" when we mean DNS-SD
- Smurf-names the functions that are already namespaced appropriately.

Reply:
@kkasperczyk-no has already prepared a PR which changes `mdns` names to `dnssd`, but we want to contribute these DNS-SD clean up PRs one at a time not to introduce too many conflicts.

Reply:
Great, thanks